appveyor.yml

```diff
@@ -0,0 +1,60 @@
+version: 5.1.0.{build}
+image: Visual Studio 2017
+environment:
+  matrix:
+  - DB: SqlServer2008
+    CONNECTION_STRING: Server=(local)\SQL2017;User ID=sa;Password=Password12!;initial catalog=nhibernate;
+  - DB: PostgreSQL
+    CONNECTION_STRING: Host=localhost;Port=5432;Username=postgres;Password=Password12!;Database=nhibernate;Enlist=true;
+  - DB: Firebird
+  - DB: MySQL
+    CONNECTION_STRING: Server=127.0.0.1;Uid=root;Pwd=Password12!;Database=nhibernate;Old Guids=True;
+init:
+    # Required for having windows endlines in sources zip
+    - git config --global core.autocrlf true
+matrix:
+  allow_failures:
+  - DB: MySQL
+build: off
+before_test:
+- ps: |-
+    switch ($env:DB) {
+      'Firebird' {
+        $FireBirdPath = 'C:\firebird'
+        # Install Firebird
+        New-Item -ItemType Directory -Force $FireBirdPath > $null
+        Push-Location $FireBirdPath
+        Invoke-WebRequest 'https://github.com/FirebirdSQL/firebird/releases/download/R3_0_3/Firebird-3.0.3.32900-0_x64.zip' -OutFile firebird.zip
+        Unblock-File firebird.zip
+        7z x firebird.zip
+        New-Item -ItemType Directory -Force Data
+        # Declare nhibernate db
+        Add-Content -Path '.\databases.conf' -Value "`r`nnhibernate = $FireBirdPath\Data\nhibernate.fdb"
+        # Minimal db settings
+        Add-Content -Path '.\firebird.conf' -Value "`r`nAuthServer = Srp`r`nAuthClient = Srp`r`nUserManager = Srp`r`nWireCrypt = Enabled"
+        # Create SYSDBA account
+        New-Item SYSDBA.sql -Type File
+        Add-Content -Path '.\SYSDBA.sql' -Value "CREATE USER SYSDBA PASSWORD 'masterkey';`r`nCOMMIT;`r`nQUIT;"
+        .\isql -user sysdba employee -input SYSDBA.sql
+        # Start Firebird
+        .\firebird.exe -a
+        Pop-Location
+      }
+      'MySQL' {
+        Start-Service 'MySQL57'
+        # Create nhibernate database (not handled by NHibernate.TestDatabaseSetup.dll)
+        $env:MYSQL_PWD = 'Password12!'
+        & 'C:\Program Files\MySQL\MySQL Server 5.7\bin\mysql' -e 'CREATE DATABASE nhibernate;' --user=root
+      }
+      'Odbc' { Start-Service 'MSSQL$SQL2017' }
+      'PostgreSQL' {
+        # Enable prepared transactions
+        Add-Content -Path 'C:\Program Files\PostgreSQL\10\data\postgresql.conf' -Value "`r`nmax_prepared_transactions = 100"
+        Start-Service 'postgresql-x64-10'
+      }
+      'SqlServer2008' { Start-Service 'MSSQL$SQL2017' }
+      'SqlServer2012' { Start-Service 'MSSQL$SQL2017' }
+    }
+test_script:
+- cmd: powershell -noprofile -command "& ./build.ps1 -TaskList Set-Configuration,Test -properties @{\"Database\" = \"%DB%\";\"ConnectionString\"=\"%CONNECTION_STRING%\"}"
```

Comment:
Travis does not support test results at all. It has an ability to upload artifacts to a third party storage provider (AWS).

Reply:
I've implemented the upload test results to AppVeyor. However they are still unusable. As AppVeyor just hangs when you try to open "Tests" tab.

Reply:
If `dotnet test` is used then the real-time reporting of test results to AppVeyor can use `Appveyor.TestLogger` (<https://github.com/Faizan2304/LoggerExtensions>).

Reply:
> I've implemented the upload test results to AppVeyor. However they are still unusable. As AppVeyor just hangs when you try to open "Tests" tab (at least on my machine).

Or maybe at least with Chrome. I am used to AppVeyor causing Chrome to hang. But Firefox (at least since its "Quantum" version) seems to better support AppVeyor pages when they have a lot of content. Currently I have opened the test tab with Chrome and Firefox, Firefox did it faster but Chrome finally did it too.

Reply:
> If dotnet test is used then the real-time reporting of test results to AppVeyor can use Appveyor.TestLogger (https://github.com/Faizan2304/LoggerExtensions).

Would not this require tweaking the platform agnostic build.ps1 script for AppVeyor? I guessed that adding an AppVeyor specific script afterward is then more compliant with Alexander's goals than handling some AppVeyor specific case for test reporting inside the general build/test script.

Reply:
`dotnet test` is broken due to the bug with `GetEntryAssembly`.